### PR TITLE
Fix bashisms

### PR DESCRIPTION
**Pull request by darko-poljak**
Hi!
This is related to https://github.com/ungleich/cdist/issues/423.
I ran checkbashisms for cdist/conf/type explorer, gencode-local, gencode-remote and manifest scripts.
This PR is the result of found bashisms removal.
This is necessary if the code should run in posix shell like dash.

The most part of it is because of using '==' instead of '=' comparison operator, suggestions to add shebang and some usage of bash specific parameter expansions like substring expansion.

### CODE REVIEW

File: cdist/conf/type/__pacman_conf/manifest [hunk 1]
```diff
@@ -42,11 +42,12 @@ contains_element() {
     MATCH=0
 
     target=$1
-    keys="${@:2}"
+    # keys="${@:2}"
```
Comment (telmich) [inline cdist/conf/type/__pacman_conf/manifest]: Maybe remove the commented out version? :-)

File: cdist/conf/type/__pacman_conf/manifest [hunk 2]
```diff
@@ -42,11 +42,12 @@ contains_element() {
...
+    shift
 
-
-    for key in ${keys}; do
-        if [ "${key}" == "${target}" ]; then
+    # for key in ${keys}; do
```
Comment (telmich) [inline cdist/conf/type/__pacman_conf/manifest]: Same here

**Comment by telmich** [comment]
Nice PR - just remove the old cruft and then we're ready for merge!

**Comment by darko-poljak** [comment]
Commented old code is removed.

**Pull request merged**